attestation-service/verifier/src/az_snp_vtpm/mod.rs

```diff
@@ -34,20 +34,25 @@ pub struct AzSnpVtpm;
 impl Verifier for AzSnpVtpm {
     async fn evaluate(
         &self,
-        nonce: String,
-        attestation: &Attestation,
+        evidence: &[u8],
+        report_data: Option<&[u8]>,
+        init_data_hash: Option<&[u8]>,
     ) -> Result<TeeEvidenceParsedClaim> {
-        let evidence = serde_json::from_str::<Evidence>(&attestation.tee_evidence)
+        if report_data.is_none() {
+            bail!("Azure SNP vTPM verifier must provide report data field!");
+        }
+
+        let report_data = report_data.expect("unexpected empty report data");
```

Comment:
those checks look redundant, I think we can extract the value in a single let-else expression.

```suggestion
    let Some(report_data) = report_data else {
        panic!("unexpected empty report data")
    };
```

Reply:
thanks, let me fix this

Reply:
Thanks for this, let me fix.

Reply:
btw, it shouldn't panic! 😅 bail! is fine